cdap-kafka/src/main/java/co/cask/cdap/kafka/run/KafkaServerMain.java

```diff
@@ -136,29 +142,24 @@ public void destroy() {
     // Nothing to do
   }
 
-  private Properties generateKafkaConfig(int brokerId, String zkConnectStr, String hostname, int port,
-                                         int numPartitions, int replicationFactor, String logDir) {
-    Preconditions.checkState(port > 0, "Port number is invalid.");
-    Preconditions.checkState(numPartitions > 0, "Num partitions should be greater than zero.");
+  private int tryParse(String val) {
+    try {
+      return Integer.parseInt(val, 10);
+    } catch (Exception e) {
+      return -1;
+    }
+  }
 
+  private Properties generateKafkaConfig(CConfiguration cConf) {
     Properties prop = new Properties();
-    prop.setProperty("broker.id", Integer.toString(brokerId));
-    if (hostname != null) {
-      prop.setProperty("host.name", hostname);
+
+    Map<String, String> propConfigs = cConf.getValByRegex("^(kafka\\.server\\.)");
+    for (Map.Entry<String, String> pair : propConfigs.entrySet()) {
+      String key = pair.getKey();
+      String trimmedKey = key.substring(13);
+      prop.setProperty(trimmedKey, pair.getValue());
+      LOG.warn("got key " + trimmedKey + " with value " + pair.getValue());
     }
-    prop.setProperty("port", Integer.toString(port));
-    prop.setProperty("socket.send.buffer.bytes", "1048576");
```

Comment:
We used to define default values for these three parameters (what used to be lines 150-152).
It seems like we no longer set these values. Is there a reason we are no longer setting them?